Library/ViewModels/AddNewCardViewModel.swift

```diff
@@ -288,8 +312,34 @@ public final class AddNewCardViewModel: AddNewCardViewModelType, AddNewCardViewM
   public let rememberThisCardToggleViewControllerIsOn: Signal<Bool, Never>
   public let saveButtonIsEnabled: Signal<Bool, Never>
   public let setStripePublishableKey: Signal<String, Never>
+  public let unsupportedCardBrandErrorText: Signal<String, Never>
   public let zipcodeTextFieldBecomeFirstResponder: Signal<Void, Never>
 
   public var inputs: AddNewCardViewModelInputs { return self }
   public var outputs: AddNewCardViewModelOutputs { return self }
 }
+
+private func cardBrandIsSupported(project: Project?, cardBrand: GraphUserCreditCard.CreditCardType) -> Bool {
+  let supportedCardBrands: [GraphUserCreditCard.CreditCardType] = [
+    .amex,
+    .diners,
+    .discover,
+    .jcb,
+    .mastercard,
+    .unionPay,
+    .visa
+  ]
+
+  guard let project = project else {
+    return supportedCardBrands.contains(cardBrand)
+  }
+
+  guard let availableCardTypes = project.availableCardTypes else {
```

Comment:
I think we should group this logic together with line 333:

```
guard let availableCardTypes = project?.availableCardTypes else {
  return supportedCardBrands.contains(cardBrand)
}
```

That will cover both the `project` being `nil` and `availableCardTypes` being `nil`.